pkg/workloadapi/workload.proto

```diff
@@ -0,0 +1,101 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package proto;
+option go_package="pkg/workloadapi";
+
+message Workload {
+  // Name represents the name for the workload. This may be elided as an optimization
+  string name = 1;
+  // Namespace represents the namespace for the workload. This may be elided as an optimization
+  string namespace = 8;
+
+  string canonical_name = 14;
+  string canonical_revision = 15;
+  string workload_type = 16;
+  string workload_name = 17;
+
+  // Address represents the IPv4/IPv6 address for the workload.
+  // This should be globally unique.
+  // This should not have a port number.
+  // TODO: Add network as discriminator
+  string address = 2;
+
+  // The SPIFFE identity of the workload.
+  string identity = 3;
+
+  // The protocol that should be used to connect to this workload.
+  Protocol protocol = 4;
+
+  // The address this workload can be reached at.
+  // If not set, address should be reached directly
+  string gateway_address = 5;
```

Comment:
I don't know if we need this. If the pod needs to be reached via gateway  - just send the list of address ( and it may be more than one ). If the pod is reached via waypoint - that can also go into address field.

Reply:
I don't mind too much keeping them separated - but like for address, please list ( dual stack, multi-nic), and I would stop the decimal and ascii address.

Also please add the 'network' field.